rgo/utils.py

```diff
@@ -114,14 +115,23 @@ def prepare_spec(spec, version, release, prefix, patches):
             match = patch_apply_re.match(line)
             if match and patches == PatchesAction.drop:
                 continue
-        elif line.startswith("%autopatch"):
+        elif line.startswith("%autopatch") or line.startswith("%patchlist"):
```

Comment:
I think `line.startswith("%patchlist")` should never be true here. Such a line will be covered by previous `elif line.startswith("%patch"):` branch.

Reply:
It does match on the previous `elif` but it then also checks the `patch_apply_re` regex and I think that won't match (because it expects at least one digit) so it can be true here.

Reply:
Once an `elif line.startswith("%patch")` branch is matched, its block is executed and none of the following branches are tested.

The execution will continue either after the `if / elif` statement on line 125, or (in case `if match and patches == PatchesAction.drop:` passes) the rest of current `for` cycle iteration is skipped and the program jumps to the next one.

Reply:
oh :man_facepalming: of course, sorry.
I have moved the more specific branch above.